tests/scripts/all.sh

```diff
@@ -3505,6 +3505,13 @@ component_build_armcc () {
     # ARM Compiler 6 - Target ARMv8.2-A - AArch64
     armc6_build_test "-O1 --target=aarch64-arm-none-eabi -march=armv8.2-a+crypto"
 }
+support_build_armcc () {
+    armc5_cc="$ARMC5_BIN_DIR/armcc"
+    armc6_cc="$ARMC6_BIN_DIR/armclang"
+    # check if Arm Compiler is installed and license server is available.
+    ("$armc5_cc" --help > /dev/null 2>&1) && \
```

Comment:
Unfortunately, in some cases, this loops for a very long time if no license server is available. E.g. if I have `$ARMLMD_LICENSE_FILE` pointing to an Arm internal machine, but I'm not connected to the Arm network, then I see
```
armcc --help
Warning: C9564I: Retrying - Fri Feb 24 10:12:06 2023

License server machine is down or not responding.
(…)
```
IIRC there's a command line option (or combination of options) to tell it to just check the license quickly. Can you please check? And same for v6, I think the license-related options are the same but I'm not sure.

Reply:
Looking through the reference guide, I only found `--vsn` which seems to be related to the license checking, but it didn't make much difference. If the address of the license server is provided, `armcc` always waits for the server becoming available for some time.

Reply:
On a second thought, I think we'd better not skip the case if the license server is not available, it should be regarded as an **error**. Thinking of the usage on CI, if the license server was down for some reason, the CI wouldn't generate a job for this case and wouldn't report an error (fix me if I am wrong). It would be difficult for us to discover that `build_armcc` was not tested.

Reply:
Maybe checking the tool path is enough. We could treat `invalid license` as an error. If we skip this test because of the invalid license, the `pre-test-checks` would always fail if the license server was down, and all other tests would not run until the license server was available again.

Reply:
Good point about the CI: if we check the license, then a license server error would cause the whole build not to run. And on user machines, this can make `--list` annoyingly slow if a license server is configured but not available.

So please reverse this change. Sorry for requesting it.